### ISSUE TITLE
REST API integration

### ISSUE THREAD

**Issue by vesnindev** [wg-easy/wg-easy#416]
Hello. Is there any possibility to: 

-- Delete device\user
-- Add device\user
-- Get QR_CODE
-- Get ConfigFile
-- disable a profile

 via API from my App ??

**Comment by eandersons** [comment]
It seems that there is API: [`src/www/js/api.js`](/WeeJeWel/wg-easy/blob/master/src/www/js/api.js), but it does not seem to be documented, so you will have to figure out how to use it on your own.
I suspect that authentication could be the biggest obstacle.

Some usage examples you can find by [searching for `api`](/WeeJeWel/wg-easy/search?q=api) in the project.
For example: 

- to dowload QR code: [`/api/wireguard/client/${client.id}/qrcode.svg`](/WeeJeWel/wg-easy/blob/faf00d46d9a15ffa519cc87a9eedc36ed28c4ddf/src/www/index.html#L248);
- to download configuration: [`/api/wireguard/client/${client.id}/configuration'`](/WeeJeWel/wg-easy/blob/faf00d46d9a15ffa519cc87a9eedc36ed28c4ddf/src/www/index.html#L257).

**Comment by megoRU** [comment]
I made wrapper for wg-easy: https://github.com/megoRU/wg-easy-wrapper

**Comment by MachineSaver** [comment]
> 

Yeah I tried authentication all the ways I can think of and it seems to not allow to connect to a session.

**Comment by Nezteb** [comment]
> I suspect that authentication could be the biggest obstacle.

[It looks like you just need to `POST` to `/session` with a JSON body that contains a `password` key](https://github.com/wg-easy/wg-easy/blob/e7f4cc2d56fe83e629166795db432ea203e635fd/src/www/js/api.js#L46-L52), though that doesn't feel super secure. It'd be nice if `wg-easy` supported token/OAuth2/OIDC auth.

It'd be useful if `wg-easy` had a documented API with either a Postman collection or an OpenAPI specifiction, simlar to https://github.com/Mawthuq-Software/Wireguard-Manager-and-API.

I'd like to build a Discord bot using https://github.com/sapphiredev/framework that lets approved users automatically add themselves to a Wireguard network; I _think_ I can do this with the current undocumented `wg-easy` API in addition to the pieces @eandersons mentioned, but we'll see how that goes. 😅

**Comment by MachineSaver** [comment]
I’ve got it figured out. Here is a gist that worked:wireguard python APIgist.github.comMessage ID: ***@***.***>

**Comment by valentin2105** [comment]
I created an enhanced version of @MachineSaver gist which is able to get QR Code and Configuration file : 

https://gist.github.com/valentin2105/70786b181fa1b972313837325b8c1d14

**Comment by peterlewis** [comment]
APIs are good. Would love to keep this on the radar.

**Comment by gaby** [comment]
@peterlewis Thank you! Probably worth looking into adding Swagger. That should help automate some of this

**Comment by gaby** [comment]
@peterlewis The stale-bot is too aggressive 😂 barely been a month and its trying to close the issue. It should be at least 90 days

**Comment by peterlewis** [comment]
Agreed. Will amend.

**Comment by pheiduck** [comment]
> @peterlewis The stale-bot is too aggressive 😂 barely been a month and its trying to close the issue. It should be at least 90 days

@gaby We could also move it to the discussion and pin it there under "Ideas".

**Comment by peterlewis** [comment]
Have set `days-before-issue-{stale,close}` to match PRs.
